Crash when setting reflect value when parsing struct

```
geppetto on  task/fix-ordering-events via 🐹 v1.21.5 on ☁️  manuel (us-east-1) 
❯ pinocchio openai  list-engines
panic: reflect.Set: value of type map[string]string is not assignable to type map[settings.ApiType]string

goroutine 1 [running]:
reflect.Value.assignTo({0x126dac0?, 0xc00199e3c0?, 0xc002806c68?}, {0x14438fc, 0xb}, 0x126dca0, 0x0)
	/usr/local/go/src/reflect/value.go:3307 +0x289
reflect.Value.Set({0x126dca0?, 0xc0018fc710?, 0x1904fe8?}, {0x126dac0?, 0xc00199e3c0?, 0x126dca0?})
	/usr/local/go/src/reflect/value.go:2260 +0xe6
github.com/go-go-golems/glazed/pkg/cmds/parameters.(*ParsedParameters).setWildcardValues(0x12c7860?, {0x126dca0?, 0xc0018fc710?, 0x1227242?}, {0x1227276, 0x9}, 0x4a?)
	/home/manuel/code/wesen/corporate-headquarters/glazed/pkg/cmds/parameters/initialize-struct.go:204 +0x189
github.com/go-go-golems/glazed/pkg/cmds/parameters.(*ParsedParameters).InitializeStruct(0xc0020ac100, {0x1263d60, 0xc0018fc710})
	/home/manuel/code/wesen/corporate-headquarters/glazed/pkg/cmds/parameters/initialize-struct.go:161 +0x3bd
github.com/go-go-golems/glazed/pkg/cmds/layers.(*ParsedLayer).InitializeStruct(...)
	/home/manuel/code/wesen/corporate-headquarters/glazed/pkg/cmds/layers/parsed-layer.go:91
github.com/go-go-golems/glazed/pkg/cmds/layers.(*ParsedLayers).InitializeStruct(0xc00121a7c8?, {0x14437ff, 0xb}, {0x1263d60, 0xc0018fc710})
	/home/manuel/code/wesen/corporate-headquarters/glazed/pkg/cmds/layers/parsed-layer.go:164 +0xa5
github.com/go-go-golems/geppetto/cmd/pinocchio/cmds/openai.(*ListEnginesCommand).RunIntoGlazeProcessor(0x0?, {0x18f68f8, 0xc001154200}, 0xc000072038?, {0x18ef9a8, 0xc000940000})
	/home/manuel/code/wesen/corporate-headquarters/geppetto/cmd/pinocchio/cmds/openai/openai.go:98 +0xe5
github.com/go-go-golems/glazed/pkg/cli.BuildCobraCommandFromGlazeCommand.func1({0x18f68f8, 0xc001154200}, 0xc0020ac050)
	/home/manuel/code/wesen/corporate-headquarters/glazed/pkg/cli/cobra.go:323 +0x108
github.com/go-go-golems/glazed/pkg/cli.BuildCobraCommandFromCommandAndFunc.func1(0xc000005b00, {0x268f2a0?, 0x0, 0x0})
	/home/manuel/code/wesen/corporate-headquarters/glazed/pkg/cli/cobra.go:184 +0xf23
github.com/spf13/cobra.(*Command).execute(0xc000005b00, {0x268f2a0, 0x0, 0x0})
	/home/manuel/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987 +0xaa3
github.com/spf13/cobra.(*Command).ExecuteC(0x24ff340)
	/home/manuel/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/manuel/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
main.main()
	/home/manuel/code/wesen/corporate-headquarters/geppetto/cmd/pinocchio/main.go:84 +0x551


```

--- a/pkg/cmds/parameters/initialize-struct.go
+++ b/pkg/cmds/parameters/initialize-struct.go
@@ -201,7 +201,7 @@ func (p *ParsedParameters) setWildcardValues(dst reflect.Value, pattern string, 
 	keyType := dst.Type().Key()
 
 	if dst.IsNil() {
-		dst.Set(reflect.MakeMapWithSize(reflect.MapOf(reflect.TypeOf(""), elemType), 0))
+		dst.Set(reflect.MakeMapWithSize(reflect.MapOf(keyType, elemType), 0))
 	}
 
 	err := p.ForEachE(func(paramName string, parameter *ParsedParameter) error {

